core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -164,104 +166,120 @@ protected void doPost(HttpServletRequest httpServletRequest,
+//                geneticProfileIdSet, sampleSetId, sampleIds, httpServletRequest);
+            boolean errorsExist = false;
+            
+            // Query single / multiple studies?
+            // --> Single study
+            // -----> regular study
+            // -----> virtual study
+            // --> Multiple studies
+            // -----> always build temporary cohort
+            String cancerStudyId = httpServletRequest.getParameter(CANCER_STUDY_ID);
+            String cancerStudyList = httpServletRequest.getParameter(CANCER_STUDY_LIST);
+            Boolean _isVirtualStudy = false;
+            if (cancerStudyId != null || cancerStudyList != null) {
+                // is single study
+                System.out.println("cancerStudyId: " + cancerStudyId);
+                System.out.println("cancerStudyList: " + cancerStudyList);
+                if (!cancerStudyId.equals("all")) {
+                    CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByStableId(cancerStudyId);
+                    // is virtual study
+                    if (cancerStudy == null) _isVirtualStudy = true;
+                    // is regular study
+                    else _isVirtualStudy = false;
+                    httpServletRequest.setAttribute(CANCER_STUDY_ID, cancerStudyId);
+                    httpServletRequest.setAttribute(CANCER_STUDY_LIST, null);
+                // multi studies
+                } else {
+                    _isVirtualStudy = true;
+                    httpServletRequest.setAttribute(CANCER_STUDY_ID, "all");
+                    httpServletRequest.setAttribute(CANCER_STUDY_LIST, cancerStudyList);
+                }
+                System.out.println("_isVirtualStudy: " + _isVirtualStudy);
+                httpServletRequest.setAttribute(IS_VIRTUAL_STUDY, _isVirtualStudy);
+                httpServletRequest.setAttribute(CASE_SET_ID, httpServletRequest.getParameter(CASE_SET_ID));
+                httpServletRequest.setAttribute(CASE_IDS,
+                    (httpServletRequest.getParameter(CASE_IDS)).replaceAll("\\\\n", "\n").replaceAll("\\\\t", "\t"));
+            }
+            
+            // Dispatch to query result page
+            if (action != null && action.equals(ACTION_SUBMIT) && (!errorsExist)) {
+                CohortDetails cohortDetails;
+                if (httpServletRequest.getParameter(CANCER_STUDY_ID).equals("all") &&
+                    httpServletRequest.getParameter(CANCER_STUDY_LIST) != null) {
+                    cohortDetails = new CohortDetails(
+                        httpServletRequest.getParameter(CANCER_STUDY_LIST).split(","),
+                        _isVirtualStudy
+                    );
+                } else {
+                    String[] _arr = new String[1];
+                    _arr[0] = httpServletRequest.getParameter(CANCER_STUDY_ID);
+                    cohortDetails = new CohortDetails(_arr, _isVirtualStudy);
```

Comment:
May be `new CohortDetails(new String[]{httpServletRequest.getParameter(CANCER_STUDY_ID)}, _isVirtualStudy)`